src/frequenz/sdk/timeseries/_voltage_streaming.py

```diff
@@ -0,0 +1,191 @@
+# License: MIT
+# Copyright © 2023 Frequenz Energy-as-a-Service GmbH
+
+"""Fetch and stream the phase-to-neutral voltage from a source component.
+
+Each phase of the phase-to-neutral voltage is fetched from the source
+component individually and then streamed as a 3-phase sample.
+"""
+
+from __future__ import annotations
+
+import asyncio
+import logging
+from typing import TYPE_CHECKING
+
+from frequenz.channels import Receiver, Sender
+
+from ..actor import ChannelRegistry
+from ..microgrid import connection_manager
+from ..microgrid.component import Component, ComponentCategory, ComponentMetricId
+from ..timeseries._base_types import Sample3Phase
+from ..timeseries._quantities import Voltage
+
+if TYPE_CHECKING:
+    # Imported here to avoid a circular import.
+    from ..actor import ComponentMetricRequest
+
+_logger = logging.getLogger(__name__)
+
+
+class VoltageStreaming:
+    """Fetch and stream the phase-to-neutral voltage from a source component.
+
+    Example:
+        ```python
+        from datetime import timedelta
+
+        from frequenz.sdk import microgrid
+        from frequenz.sdk.timeseries import ResamplerConfig
+
+        await microgrid.initialize(
+            "127.0.0.1",
+            50051,
+            ResamplerConfig(resampling_period=timedelta(seconds=1))
+        )
+
+        # Get a receiver for the phase-to-neutral voltage.
+        voltage_recv = microgrid.voltage().new_receiver()
+
+        async for voltage_sample in voltage_recv:
+            print(voltage_sample)
+        ```
+    """
+
+    def __init__(
+        self,
+        resampler_subscription_sender: Sender[ComponentMetricRequest],
+        channel_registry: ChannelRegistry,
+        source_component: Component | None = None,
+    ):
+        """Initialize the phase-to-neutral voltage streaming.
+
+        Args:
+            resampler_subscription_sender: The sender for sending metric
+                requests to the resampling actor.
+            channel_registry: The channel registry for the phase-to-neutral
+                voltage streaming.
+            source_component: The source component to receive the
+                phase-to-neutral voltage. If None, it fetches the source
+                component from the connection manager.
+
+        """
+        self._resampler_subscription_sender = resampler_subscription_sender
+        """The sender for sending metric requests to the resampling actor."""
+
+        self._channel_registry = channel_registry
+        """The channel registry for the phase-to-neutral voltage streaming."""
+
+        if not source_component:
+            component_graph = connection_manager.get().component_graph
+            source_component = component_graph.find_first_descendant_component(
+                root_category=ComponentCategory.GRID,
+                descendant_categories=[
+                    ComponentCategory.METER,
+                    ComponentCategory.EV_CHARGER,
```

Comment:
So in this case you are really purposely skipping inverters?

Reply:
Yes, I deliberately skipped it because voltage is not available in inverter data methods from Microgrid API source
(see https://github.com/frequenz-floss/frequenz-sdk-python/blob/v1.x.x/src/frequenz/sdk/actor/_data_sourcing/microgrid_api_source.py#L60-L78)
But  I can't tell if they are missing or they just shouldn't be there. @sahas-subramanian-frequenz maybe you know the reason?

Reply:
I think inverters have to be a possible source as well.  Could you update the data sourcing actor and the underlying `InverterData` types as well @daniel-zullo-frequenz?

In this PR or you could make an issue to deal with it separately, up to you.

(also I'm not using the `sahas-subramanian-frequenz` account anymore.  Could you please tag @shsms instead?  I guess I have to look into maybe renaming the old account or something)

Reply:
Added 4 small commits to address this suggestion